store.go

```diff
@@ -369,3 +377,71 @@ func (c *controller) processEndpointUpdate(ep *endpoint) bool {
 
 	return false
 }
+
+func (c *controller) validateLocalDatastoreConfig() bool {
+	return c.cfg != nil && c.cfg.LocalDataStore.Client.Provider != "" && c.cfg.LocalDataStore.Client.Address != ""
+}
+
+func (c *controller) getNetworksFromLocalStore() ([]*store.KVPair, error) {
+	c.Lock()
+	cs := c.localStore
+	c.Unlock()
+	return cs.KVStore().List(datastore.Key(datastore.NetworkKeyPrefix))
+}
+
+func (c *controller) getDataStore(scope driverapi.Scope) (datastore datastore.DataStore) {
+	c.Lock()
+	if scope == driverapi.GlobalScope {
+		datastore = c.store
+	} else if scope == driverapi.LocalScope {
+		datastore = c.localStore
+	}
+	c.Unlock()
+	return
+}
+
+func (c *controller) getEndpointsFromLocalStore() ([]*store.KVPair, error) {
+	c.Lock()
+	cs := c.localStore
+	c.Unlock()
+	return cs.KVStore().List(datastore.Key(datastore.EndpointKeyPrefix))
+}
+
+func (c *controller) processEndpointsUpdate(eps []*store.KVPair, prune *endpointTable) {
+	for _, epe := range eps {
+		var ep endpoint
+		err := json.Unmarshal(epe.Value, &ep)
+		if err != nil {
+			log.Error(err)
+			continue
+		}
+		if prune != nil {
+			delete(*prune, ep.id)
+		}
+		ep.SetIndex(epe.LastIndex)
+		if nid, err := networkIDFromKey(epe.Key); err != nil {
```

Comment:
Please look at the next comment. Lets remove this function and use the `endpoint.NetworkIDFromKey` function